servlet/src/main/java/com/linecorp/armeria/server/servlet/ServletRegistration.java

```diff
@@ -0,0 +1,252 @@
+    @Nullable
+    private ServletSecurityElement servletSecurityElement;
+    @Nullable
+    private String roleName;
+
+    /**
+     * Creates a new instance.
+     */
+    public ServletRegistration(String servletName, Servlet servlet, DefaultServletContext servletContext,
+                               UrlMapper<ServletRegistration> urlMapper) {
+        this(servletName, servlet, servletContext, urlMapper, ImmutableMap.copyOf(new HashMap<>()));
+    }
+
+    /**
+     * Creates a new instance.
+     */
+    public ServletRegistration(String servletName, Servlet servlet, DefaultServletContext servletContext,
+                               UrlMapper<ServletRegistration> urlMapper, Map<String, String> initParameterMap) {
+        requireNonNull(servletName, "servletName");
+        requireNonNull(servlet, "servlet");
+        requireNonNull(servletContext, "servletContext");
+        requireNonNull(urlMapper, "urlMapper");
+
+        this.servletName = servletName;
+        this.servlet = servlet;
+        this.servletContext = servletContext;
+        this.urlMapper = urlMapper;
+        this.initParameterMap = ImmutableMap.copyOf(initParameterMap);
+        servletConfig = new ServletConfig() {
+            @Override
+            public String getServletName() {
+                return ServletRegistration.this.servletName;
+            }
+
+            @Override
+            public ServletContext getServletContext() {
+                return ServletRegistration.this.servletContext;
+            }
+
+            @Override
+            @Nullable
+            public String getInitParameter(String name) {
+                return ServletRegistration.this.getInitParameter(name);
+            }
+
+            @Override
+            public Enumeration<String> getInitParameterNames() {
+                return Collections.enumeration(ImmutableSet.copyOf(getInitParameters().keySet()));
+            }
+        };
+    }
+
+    /**
+     * Get servlet security element.
+     */
+    @Nullable
+    public ServletSecurityElement getServletSecurityElement() {
+        return servletSecurityElement;
+    }
+
+    /**
+     * Get multipart config element.
+     */
+    @Nullable
+    public MultipartConfigElement getMultipartConfigElement() {
+        return multipartConfigElement;
+    }
+
+    /**
+     * Get servlet config.
+     */
+    public ServletConfig getServletConfig() {
+        return servletConfig;
+    }
+
+    /**
+     * Get servlet.
+     */
+    public Servlet getServlet() {
+        return servlet;
+    }
+
+    /**
+     * Is async supported.
+     */
+    public Boolean isAsyncSupported() {
+        return asyncSupported;
+    }
+
+    /**
+     * Get load on startup.
+     */
+    public int getLoadOnStartup() {
+        return loadOnStartup;
+    }
+
+    /**
+     * Is initialization servlet cas.
+     */
+    public boolean isInitServletCas(boolean expect, boolean update) {
```

Comment:
Can we remove this method and make this registration is initialized when it created?